cmd/root.go

```diff
@@ -214,6 +243,48 @@ func initConfig() {
 	rootConfig.Debug = (viper.GetInt("verbose") >= 1)
 	rootConfig.Timeout = (viper.GetInt("timeout"))
 
+	clientCertContent := ""
+
+	if (viper.GetString("tls-client-cert") == "" && viper.GetString("tls-client-key") != "") ||
+		(viper.GetString("tls-client-cert") != "" && viper.GetString("tls-client-key") == "") {
+		fmt.Printf("tls-client-cert and tls-client-key must be used in conjunction")
+		os.Exit(1)
+	}
+
+	if (viper.GetString("tls-client-cert-file") == "" && viper.GetString("tls-client-key-file") != "") ||
+		(viper.GetString("tls-client-cert-file") != "" && viper.GetString("tls-client-key-file") == "") {
+		fmt.Printf("tls-client-cert-file and tls-client-key-file must be used in conjunction")
+		os.Exit(1)
+	}
```

Comment:
This code is really hard to read and reason about.
Here is an alternative:
1. Read certificate first (from the flag and then second priority to the file-based flag) and store in a variable.
2. Do the same for the key
3. If cert or key are exclusively empty, return an error.

Reply:
Ok will update accordingly.